libwav/files/sndwav.c

```diff
@@ -0,0 +1,361 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <malloc.h>
+
+#include <kos/thread.h>
+#include <dc/sound/stream.h>
+
+#include "sndwav.h"
+#include "libwav.h"
+
+/* Keep track of things from the Driver side */
+#define SNDDRV_STATUS_NULL         0x00
+#define SNDDRV_STATUS_READY        0x01
+#define SNDDRV_STATUS_DONE         0x02
+
+/* Keep track of things from the Decoder side */
+#define SNDDEC_STATUS_NULL         0x00
+#define SNDDEC_STATUS_READY        0x01
+#define SNDDEC_STATUS_STREAMING    0x02
+#define SNDDEC_STATUS_PAUSING      0x03
+#define SNDDEC_STATUS_STOPPING     0x04
+#define SNDDEC_STATUS_RESUMING     0x05
+
+#define STREAM_BUFFER_SIZE 65536
```

Comment:
Not used

Reply:
Youre right.  When I updated to dynamic memory I ended up using SND_STREAM_BUFFER_MAX instead which has the same value.